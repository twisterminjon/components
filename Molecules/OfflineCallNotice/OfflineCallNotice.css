.offlinecallnotice-wrap {
  position: fixed;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;

  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  z-index: var(--z-callnotice);
}

.offlinecallnotice-dimmer {
  position: fixed;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  z-index: 21;

  background-color: var(--call_declined_color);
}

.offlinecallnotice-content {
  position: absolute;
  z-index: 22;
}

.offlinecallnotice-buttons {
  position: relative;
  width: 218px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.offlinecallnotice-buttons-center {
  justify-content: space-around;
  /* Note sure why the margin is required but it won't center without it */
  margin: 0 auto;
}

.offlinecallnotice-notice {
  display: block;
  margin-bottom: 20px;
  text-align: center;
  color: var(--white);
  font-size: 24px;
}

.offlinecallnotice-notice-title {
  font-weight: bold;
  font-size: 28px;
}

.offlinecallnotice-notice:last-child {
  margin-bottom: 60px;
  font-weight: 300;
  font-size: 20px;
}
