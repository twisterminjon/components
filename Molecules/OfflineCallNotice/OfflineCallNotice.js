import React, { Component } from 'react'
import PropTypes from 'prop-types'

import CancelCallButton from '../../Molecules/CancelCallButton/CancelCallButton'

import './OfflineCallNotice.css'

export default class OfflineCallNotice extends Component {
  static propTypes = {
    /** If true, will display notice */
    show: PropTypes.bool.isRequired,
    /** A function to run when the cancel call button is clicked */
    onCancel: PropTypes.func.isRequired,
    /** Function to redirect to dashboard according to Data */
    onGoToDashboard: PropTypes.func.isRequired,
  }

  render() {
    const { show, onCancel, onGoToDashboard } = this.props

    if (!show) return null

    const handleReturn = () => {
      onCancel()
      onGoToDashboard()
    }

    return (
      <div className="offlinecallnotice-wrap">
        <div className="offlinecallnotice-dimmer" />
        <div className="offlinecallnotice-content">
          <div>
            <span className="offlinecallnotice-notice offlinecallnotice-notice-title">Oops... Call is over.</span>

            <span className="offlinecallnotice-notice">
              Something happened to the connection and the other person is no longer on the call.
            </span>
          </div>
          <div className={`offlinecallnotice-buttons offlinecallnotice-buttons-center`}>
            <CancelCallButton ghost={true} onClick={handleReturn} />
          </div>
        </div>
      </div>
    )
  }
}
