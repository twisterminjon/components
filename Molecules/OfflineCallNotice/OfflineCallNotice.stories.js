import React from 'react'

import { storiesOf } from '@storybook/react'
import { withKnobs, boolean } from '@storybook/addon-knobs'

import OfflineCallNotice from './OfflineCallNotice'

const stories = storiesOf('Provider/v1/Molecules/CallNotices', module)
stories.addDecorator(withKnobs)

stories.add('OfflineCallNotice', () => {
  const showLabel = 'show'
  const showDefault = true
  const showValue = boolean(showLabel, showDefault)

  return (
    <div
      style={{
        backgroundImage: 'url("https://picsum.photos/500/500/?random")',
        backgroundRepeat: 'no-repeat',
        backgroundSize: 'cover',
        position: 'absolute',
        top: 0,
        right: 0,
        bottom: 0,
        left: 0,
      }}>
      <OfflineCallNotice
        show={showValue}
        onCancel={() => {
          alert('cancel clicked')
        }}
        onGoToDashboard={() => {
          alert('Go to Dashboard clicked')
        }}
      />
    </div>
  )
})
