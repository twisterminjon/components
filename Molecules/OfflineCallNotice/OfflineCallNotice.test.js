import React from 'react'
import OfflineCallNotice from './OfflineCallNotice'

describe('OfflineCallNotice', () => {
  const mockFunction = jest.fn()

  it('matches the snapshot when visible', () => {
    const wrapper = window.shallow(
      <OfflineCallNotice show={true} onCancel={mockFunction} onGoToDashboard={mockFunction} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('matches the snapshot when hidden', () => {
    const wrapper = window.shallow(
      <OfflineCallNotice show={false} onCancel={mockFunction} onGoToDashboard={mockFunction} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
