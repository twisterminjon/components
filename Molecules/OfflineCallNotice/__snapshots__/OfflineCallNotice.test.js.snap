// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfflineCallNotice matches the snapshot when hidden 1`] = `""`;

exports[`OfflineCallNotice matches the snapshot when visible 1`] = `
<div
  className="offlinecallnotice-wrap"
>
  <div
    className="offlinecallnotice-dimmer"
  />
  <div
    className="offlinecallnotice-content"
  >
    <div>
      <span
        className="offlinecallnotice-notice offlinecallnotice-notice-title"
      >
        Oops... Call is over.
      </span>
      <span
        className="offlinecallnotice-notice"
      >
        Something happened to the connection and the other person is no longer on the call.
      </span>
    </div>
    <div
      className="offlinecallnotice-buttons offlinecallnotice-buttons-center"
    >
      <CancelCallButton
        ghost={true}
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;
